src/gs-updates-page.c

```diff
@@ -1516,7 +1516,8 @@ gs_updates_page_upgrade_download_cb (GsUpgradeBanner *upgrade_banner,
 
 	if (self->cancellable_upgrade_download != NULL)
 		g_object_unref (self->cancellable_upgrade_download);
-	self->cancellable_upgrade_download = g_cancellable_new ();
+	self->cancellable_upgrade_download = g_object_ref (gs_app_get_cancellable (app));
```

Comment:
I didn't explain the reason a lot here because we already use a similar logic for all apps' operations. The upgrade is no exception and should use the GsApp object's cancellable too. I will send this patch upstream too.

Reply:
OK